opnieuw/test_util.py

```diff
@@ -3,13 +3,14 @@
 from collections.abc import Iterator
 from contextlib import AbstractContextManager
 
-from .retries import Action, DoCall, RetryState, replace_retry_state
+from .retries import DoWait, RetryState, replace_retry_state
 
 
 class WaitLessRetryState(RetryState):
-    def __iter__(self) -> Iterator[Action]:
+    def __iter__(self) -> Iterator[DoWait]:
         for _ in range(self.max_calls_total):
-            yield DoCall()
+            # Yield a DoWait that leads to no waiting
+            yield DoWait(0, 0, 0, 0)
```

Comment:
This `DoWait(0, 0, 0, 0)` looks like a magical object. It is pretty hard to understand how this interacts with the decorator itself. It is worth exploring whether we can prevent this by also removing the concept of `DoWait`. I think we can solely rely on `RetryState` to inform the decorator on how long to wait. Lets discuss offline.